### DISCUSSION TITLE
Get rid of `TinyConfig` custom format and use JSON instead.

### DISCUSSION THREAD

**Discussion by KOLANICH** [ihhub/fheroes2#7332] [Ideas]
IMHO using any custom format will only create interoperability issues. #3477 is related.

**Comment by ihhub** [comment]
The code for configuration settings is being tested and works as per requirements. Changing it would require additional resources for implementation and testing. On top of this many users would face problems with reset settings which is bad user experience. Changing code for the sake of perfection is great but we need to manage our time as well. Such a task is possible to do but based on our limited resources and an existing set of high priority tasks, we won't do this till we close existing issues.

**Reply by KOLANICH** [reply]
I don't understand why this custom format was introduced originally at all instead of just using JSON for which libraries exist, using which doesn't require developing an own serialization format and testing it. Is it for interoperability to original games or what?

>users would face problems with reset settings

I guess it is not such a big problem. both impls can be kept for few releases, migrating the settings into the new format.

**Reply by ihhub** [reply]
The project had been developed by another team and started at 2009. The previous team implemented this format for configuration file. I believe JSON in 2009 wasn't that popular and not many libraries exist to handle it.

**Reply by KOLANICH** [reply]
Thanks for the info.
